light/store/db/db.go

```diff
@@ -30,7 +29,7 @@ type dbs struct {
 // want to use one DB with many light clients).
 func New(db dbm.DB, prefix string) store.Store {
 
-	lightStore := &dbs{db: db, prefix: prefix}
+	lightStore := &dbs{db: dbm.NewPrefixDB(db, []byte(prefix))}
```

Comment:
If the caller wants to use a prefix, they should just pass a `PrefixDB` as `db`. It implements the `DB` interface, so it's entirely transparent and the light store won't need any special handling of it at all.